code/modules/mob/living/simple_animal/bot/medbot.dm

```diff
@@ -547,7 +549,9 @@
 					var/obj/item/storage/medkit/medkit = medkit_type
 					if(treatment_method == BRUTE && initial(medkit.damagetype_healed) == BRUTE) //specialized brute gets a bit of bonus, as a snack.
 						healies *= 1.1
-					if(bot_cover_flags & BOT_COVER_EMAGGED)
+					if(alt_heal(patient))
```

Comment:
Instead of doing this, you can make the healing be a proc like ``heal_target()`` and on the subtype you'll call parent and add a chance to inject the space cola

Reply:
wouldnt it be hard to like- check that the person is alive multiple times?

Reply:
Just the tending part would be in this new proc, it would be the healing and logging

Reply:
?? so would it be like
proc/alt_heal()
   add_reagents
   return TRUE


if(alt_heal(patient))
   log_msg

Reply:
No, it would be 

/proc/heal_patient()
if(emagged)
add_reagents
apply_damage
log_combat
else
heal_patient
log_healing

then your subtype will call parent, then do a prob to also inject with space cola